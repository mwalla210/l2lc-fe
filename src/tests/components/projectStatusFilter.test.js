import React from 'react'
import renderer from 'react-test-renderer'
import ProjectStatusFilter from '../../components/projectStatusFilter'
jest.mock('reactstrap', () => ({
  ButtonDropdown: 'ButtonDropdown',
  DropdownToggle: 'DropdownToggle',
  DropdownMenu: 'DropdownMenu',
}))
jest.mock('react-checkbox-group', () => ({
  Checkbox: 'Checkbox',
  CheckboxGroup: 'CheckboxGroup',
}))

const defaultOptions = {
  onChange: jest.fn(),
  filter:{
    value:null,
  },
  page: {
    tableModel:{
      filterDD: jest.fn(),
      toggleDropdown: jest.fn(),
    },
  },
  website: {
    currentUser:null,
  },
}

const alternateOptions = {
  onChange: jest.fn(),
  filter:null,
  page: {
    tableModel:{
      filterDD: jest.fn(),
      toggleDropdown: jest.fn(),
    },
  },
  website: {
    currentUser:null,
  },
}

describe('ProjectStatusFilter', () => {
  it ('Renders with snapshot', () => {
    let options = Object.assign({}, defaultOptions)
    const component = renderer.create(
      <ProjectStatusFilter {...options}/>,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it ('Renders with snapshot, alternate arguments', () => {
    let options = Object.assign({}, alternateOptions)
    const component = renderer.create(
      <ProjectStatusFilter {...options}/>,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Renders and calls checkChange', () => {
    let options = Object.assign({}, defaultOptions)
    const component = renderer.create(
      <ProjectStatusFilter {...options}/>,
    )
    const inst = component.getInstance()

    expect(inst.props.onChange.mock.calls.length).toBe(0)
    inst.checkChange()
    expect(inst.props.onChange.mock.calls.length).toBe(1)
  })

})
