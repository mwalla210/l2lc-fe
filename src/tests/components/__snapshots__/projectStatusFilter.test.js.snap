// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectStatusFilter Renders with snapshot 1`] = `
<ButtonDropdown
  isOpen={[MockFunction]}
  style={
    Object {
      "height": "30px",
    }
  }
  toggle={[MockFunction]}
>
  <DropdownToggle
    caret={true}
    color="secondary"
    outline={true}
    style={
      Object {
        "paddingTop": "2px",
      }
    }
  >
    Filter
  </DropdownToggle>
  <DropdownMenu>
    <CheckboxGroup
      name="Filters"
      onChange={[Function]}
      value={null}
    >
      <label
        style={
          Object {
            "marginLeft": "8px",
          }
        }
      >
        <Checkbox
          value="Received"
        />
         Received
      </label>
      <br />
      <label
        style={
          Object {
            "marginLeft": "8px",
          }
        }
      >
        <Checkbox
          value="In Progress"
        />
         In Progress
      </label>
      <br />
      <label
        style={
          Object {
            "marginLeft": "8px",
          }
        }
      >
        <Checkbox
          value="Completed"
        />
         Completed
      </label>
      <br />
    </CheckboxGroup>
  </DropdownMenu>
</ButtonDropdown>
`;

exports[`ProjectStatusFilter Renders with snapshot, alternate arguments 1`] = `
<ButtonDropdown
  isOpen={[MockFunction]}
  style={
    Object {
      "height": "30px",
    }
  }
  toggle={[MockFunction]}
>
  <DropdownToggle
    caret={true}
    color="secondary"
    outline={true}
    style={
      Object {
        "paddingTop": "2px",
      }
    }
  >
    Filter
  </DropdownToggle>
  <DropdownMenu>
    <CheckboxGroup
      name="Filters"
      onChange={[Function]}
      value={Array []}
    >
      <label
        style={
          Object {
            "marginLeft": "8px",
          }
        }
      >
        <Checkbox
          value="Received"
        />
         Received
      </label>
      <br />
      <label
        style={
          Object {
            "marginLeft": "8px",
          }
        }
      >
        <Checkbox
          value="In Progress"
        />
         In Progress
      </label>
      <br />
      <label
        style={
          Object {
            "marginLeft": "8px",
          }
        }
      >
        <Checkbox
          value="Completed"
        />
         Completed
      </label>
      <br />
    </CheckboxGroup>
  </DropdownMenu>
</ButtonDropdown>
`;
